components/renku_data_services/session/api.spec.yaml

```diff
@@ -274,6 +274,10 @@ components:
           $ref: "#/components/schemas/Description"
         container_image:
           $ref: "#/components/schemas/ContainerImage"
+        build_parameter:
+          $ref: "#/components/schemas/BuildParameter"
+        environment_image_source:
+          $ref: "#/components/schemas/EnvironmentImageSource"
```

Comment:
For GET calls, this is OK, though not great because the types do allow non-sensical values to be sent, e.g.

```json
{
  "build_parameters": { [object] },
  "environment_image_source": "image"
}
```

or

```json
{
  <no build_parameters>,
  "environment_image_source": "build"
}
```